src/Azure.Functions.Cli/Common/DurableManager.cs

```diff
@@ -0,0 +1,362 @@
+
+        private TaskHubClient _client;
+
+        private string _taskHubName;
+
+        private readonly string _connectionStringKey;
+
+        public const string DefaultConnectionStringKey = "AzureWebJobsStorage";
+
+        public const string DefaultTaskHubName = "DurableFunctionsHub";
+
+        public const string DurableAzureStorageExtensionName = "DurableTask.AzureStorage.dll";
+
+        public const string MinimumDurableAzureStorageExtensionVersion = "1.4.0";
+
+        public DurableManager(ISecretsManager secretsManager)
+        {
+            this._secretsManager = secretsManager;
+            this.SetConnectionStringAndTaskHubName(ref _taskHubName, ref _connectionStringKey);
+        }
+
+        private void SetConnectionStringAndTaskHubName(ref string taskHubName, ref string connectionStringKey)
+        {
+            // Set connection string key and task hub name to defaults
+            connectionStringKey = DefaultConnectionStringKey;
+            taskHubName = DefaultTaskHubName;
+
+            try
+            {
+                if (File.Exists(ScriptConstants.HostMetadataFileName))
+                {
+                    // Attempt to retrieve Durable override settings from host.json
+                    dynamic hostSettings = JObject.Parse(File.ReadAllText(ScriptConstants.HostMetadataFileName));
+
+                    string version = hostSettings["version"];
+                    if (version?.Equals("2.0") == true)
+                    {
+                        // If the version is (explicitly) 2.0, prepend path to 'durableTask' with 'extensions'
+                        connectionStringKey = hostSettings?.extensions?.durableTask?.AzureStorageConnectionStringName ?? connectionStringKey;
+                        taskHubName = hostSettings?.extensions?.durableTask?.HubName ?? taskHubName;
+                    }
+                    else
+                    {
+                        connectionStringKey = hostSettings?.durableTask?.AzureStorageConnectionStringName ?? connectionStringKey;
+                        taskHubName = hostSettings?.durableTask?.HubName ?? taskHubName;
+                    }
+                }
+                else
+                {
+                    ColoredConsole.WriteLine(Yellow($"Could not find local host metadata file '{ScriptConstants.HostMetadataFileName}'"));
+                }
+            }
+            catch (Exception e)
+            {
+                ColoredConsole.WriteLine(Yellow($"Exception thrown while attempting to parse override connection string and task hub name from '{ScriptConstants.HostMetadataFileName}':"));
+                ColoredConsole.WriteLine(Yellow(e.Message));
+            }     
+        }
+
+        private void SetStorageServiceAndTaskHubClient(ref AzureStorageOrchestrationService orchestrationService, ref TaskHubClient taskHubClient, string connectionStringKey = null, string taskHubName = null)
```

Comment:
This method is executed before any of the action methods in order to initialize the `AzureStorageOrchestrationService` and `TaskHubClient`. They're `ref` parameters because they modify DurableManager's `_orchestrationService` and `_client` fields. Those fields are used later by all of the action methods.

Reply:
With the changes in the state I was reviewing, this was never assigned to within the scope of this method. So if the expectation is that this would be set somehow, then there's a bug in the method. We can follow up in person if needed.

Reply:
I see you've updated the code to assign to that value and change this to an `out` to avoid the issue that existed before. So this looks good now.